src/com/hms/flexyosisoftconnector/Tag.java

```diff
@@ -120,6 +193,417 @@ public void recordTagValue(String time)
+          Logger.LOG_DEBUG("tracker was negative, setting to current time");
+          writeTimeTrackerFile(currentTimeMs);
+          startTimeMs = currentTimeMs;
+          // else if the time period is longer then the time window but also positive
+      } else if (currentTimeMs - startTimeMs > timeWindowMilliseconds) {
+          endTime = convertTimeIntHelper((Long.parseLong(startTime) + timeWindowMilliseconds));
+          Logger.LOG_DEBUG(getTagName()+" grabbing "+ (timeWindowMilliseconds / millisecondConverter / minutesConverter) +" minutes of data: EBD generated, ");
+          Logger.LOG_DEBUG(getTagName()+" is " + howFarBehindMinutes + " minutes behind current time");
+      } else {
+          // Otherwise just set the end time to current time, because the end of the time period would of been into the future.
+          // This is different from the first case where the beginning of the time period would of been in the future
+          endTime= convertTimeIntHelper(currentTimeMs);
+          Logger.LOG_DEBUG(getTagName()+"grabbing  "+((currentTimeMs - startTimeMs) / millisecondConverter) + " seconds: EBD generating");
+      }
+
+      // format the time tracker to correct format after we have used it in the previous format
+      startTime = convertTimeIntHelper(Long.parseLong(startTime));
+
+      String tagName = getTagName();
+
+      try {
+          // https://ewonsupport.biz/ebd/ <- has a nice EBD string builder. :)
+          // Command gets historical logs from start and end time for the tag and exports in binary.
+          Exporter exporter = new Exporter("$dtHL$ftB$st" + startTime + "$et" + endTime + "$fnvalues$tn" + tagName);
+          exporter.ExportTo("file://" + tagDataFileName);
+          exporter.close();
+      }catch (IOException ioe){
+          Logger.LOG_ERR(getTagName()+" Error in export block descriptor");
+          Logger.LOG_ERR("$dtHL$ftT$st" + startTime + "$et" + endTime + "$fnvalues$tn" + tagName);
+      }
+   }
+
+   /**
+    * addToQueue is used to check if we should be reading in from the historical logs at this time.
+    * Without it, we run into issues when removing from the list in dataposter while adding to the list
+    * from the main method
+    *
+    * @param newValue, new value to set
+    * True - keep adding to queue
+    * False = stop adding to queue
+    */
+   public void setAddToQueue(boolean newValue) {
+       addToQueue = newValue;
+   }
+
+   public boolean getAddToQueue() {
+       return addToQueue;
+   }
+
+   /**
+    * Converts given long into date string of expected format
+    * and writes it to a file
+    *
+    * @param timeInt epoch time in seconds to get converted and stored in file
+    * @return just return what string it set
+    */
+   public String writeTimeTrackerFile(long timeInt)
+   {
+      // convert from epoch time long to string
+      String fileValueString = Long.toString(timeInt); //convertTimeIntHelper(timeInt);

```

Comment:
commented out code?

Reply:
I missed this. I changed how the file that stores the start time is formatted. That comment will be deleted